deisctl/cmd/cmd.go

```diff
@@ -268,18 +299,28 @@ func Install(targets []string, b backend.Backend, checkKeys func() error) error
 	return nil
 }
 
-func installDefaultServices(b backend.Backend, wg *sync.WaitGroup, outchan chan string, errchan chan error) {
+func installDefaultServices(b backend.Backend, stateless bool, wg *sync.WaitGroup, outchan chan string, errchan chan error) {
 
-	outchan <- fmt.Sprintf("Storage subsystem...")
-	b.Create([]string{"store-daemon", "store-monitor", "store-metadata", "store-volume", "store-gateway@1"}, wg, outchan, errchan)
-	wg.Wait()
+	if !stateless {
+		outchan <- fmt.Sprintf("Storage subsystem...")
+		b.Create([]string{"store-daemon", "store-monitor", "store-metadata", "store-volume", "store-gateway@1"}, wg, outchan, errchan)
+		wg.Wait()
+	}
 
 	outchan <- fmt.Sprintf("Logging subsystem...")
-	b.Create([]string{"logger", "logspout"}, wg, outchan, errchan)
+	if stateless {
+		b.Create([]string{"logspout"}, wg, outchan, errchan)
+	} else {
+		b.Create([]string{"logger", "logspout"}, wg, outchan, errchan)
```

Comment:
there seems to be a lot of if/else with duplicate code, but different component lists. What if we converted the list of strings into actual component structures that contained the component name and a boolean flag representing if the component is "stateless"? That way we can programatically remove certain components from the list based on whether it's stateless or not and reduce LoC. Would that seem fair to undertake with this PR or would a followup be better (considering that we're close to PR freeze day)?